lib/commands/skill/add-locales/helper.js

```diff
@@ -0,0 +1,148 @@
+        // copy iModel JSONs that contains same language
+        iModelSourceByLocales.forEach((v, k) => {
+            if (fs.existsSync(v)) {
+                const sourceLocale = path.basename(v, path.extname(v));
+                const targetFilePath = path.join(iModelFolderPath, `${k}.json`);
+                fs.copySync(v, targetFilePath); // do not fail if the target locale exists already
+                uniqueTemplateSet.delete(v);
+                const sourceManifestLocale = Manifest.getInstance().getPublishingLocale(sourceLocale);
+                Manifest.getInstance().setPublishingLocale(k, sourceManifestLocale);
+            }
+        });
+        // retrieve remote templates when no existing JSON shares the same language
+        _retrieveTemplatesByLanguage(uniqueTemplateSet, doDebug, (templateErr, templateByLanguage) => {
+            if (templateErr) {
+                return callback(templateErr);
+            }
+            iModelSourceByLocales.forEach((v, k) => {
+                if (!fs.existsSync(v)) {
+                    const targetFilePath = path.join(iModelFolderPath, `${k}.json`);
+                    fs.writeFileSync(targetFilePath, templateByLanguage.get(v), { encoding: 'utf-8' });
+                    Manifest.getInstance().setPublishingLocale(k, { name: 'please change' });
+                }
+            });
+            Manifest.getInstance().write();
+            callback(undefined, iModelSourceByLocales);
+        });
+    });
+}
+
+/**
+ * Based on user's selected locales, and the local project's iModel files, decide where to get the
+ * iModel file to add in user's local project. Exclude reusableLocales themselves from the result.
+ *
+ * @param {Array} selectedLocales
+ * @param {String} profile
+ * @param {Boolean} doDebug
+ * @returns {Function} callback Map { locale: filePath/remoteURI }
+ */
+function _getNewLocaleModelUri(selectedLocales, profile, doDebug, callback) {
+    const skillMetaController = new SkillMetadataController({ profile, doDebug });
+    const localIModelByLocale = skillMetaController.getInteractionModelLocales();
+    const result = new Map();
+    // get from map of templates
+    httpClient.request({
+        url: CONSTANTS.TEMPLATES.INTERACTION_MODEL_MAP,
+        method: CONSTANTS.HTTP_REQUEST.VERB.GET
+    }, 'GET_INTERACTION_MODEL_MAP', doDebug, (error, templateMapResponse) => {
+        if (error || !templateMapResponse.statusCode || templateMapResponse.statusCode !== 200) {
+            return callback('Failed to retrieve the template list. Please run again with --debug to check more details.');
+        }
+        let templateIndexMap = templateMapResponse.body;
+        if (typeof templateMapResponse.body === 'string') {
+            templateIndexMap = JSON.parse(templateMapResponse.body);
+        }
+        // assign for each locale
+        for (const locale of selectedLocales) {
+            if (R.keys(localIModelByLocale).includes(locale)) {
```

Comment:
Why do you need to use R if it can be handled by native JS methods

Reply:
I noticed this module has R imported already, so i directly use it. I think keep using R as much as possible is good thing, as `Object.keys` is longer then `R.keys` plus there could be some cascading benefits from using ramda. So I feel we don't really need to be too strict over this.

Reply:
I think it goes against the idea of reducing number of dependencies. I dont think number of characters in function name is a good metric.